sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -86,6 +87,20 @@ trait CheckAnalysis extends PredicateHelper {
   }
 
   def checkAnalysis(plan: LogicalPlan): Unit = {
+    // Analysis that needs to be performed top down can be added here.
+    plan.foreach {
+      case p if p.analyzed => // Skip already analyzed sub-plans
+
+      case alter: AlterTable =>
+        alter.table match {
+          case u @ UnresolvedTableWithViewExists(view) if !view.isTempView =>
```

Comment:
what's the error message for temp view?

Reply:
The current/existing tests expect `[t] is a temp view not a table`.

Reply:
shall we use the same error message as permanent views?

Reply:
OK. I will do this as a follow-up. Thanks @cloud-fan for the review!